crates/oxc_parser/src/lexer/mod.rs

```diff
@@ -97,7 +97,20 @@ pub struct Lexer<'a> {
 
 #[allow(clippy::unused_self)]
 impl<'a> Lexer<'a> {
-    pub fn new(allocator: &'a Allocator, source_text: &'a str, source_type: SourceType) -> Self {
+    /// Create new `Lexer`.
+    ///
+    /// # SAFETY
+    /// Caller must not allow more than one `Lexer` to exist at any one time on a thread.
+    /// In practice, this is easy to satisfy. Only means that, apart from `ParserImpl::new`,
+    /// no method of `ParserImpl` or `Lexer` can call this and create another `Lexer`.
+    #[allow(clippy::missing_safety_doc)] // Clippy is wrong!
+    pub unsafe fn new(
+        allocator: &'a Allocator,
+        source_text: &'a str,
+        source_type: SourceType,
+    ) -> Self {
+        // SAFETY: We only create a single `Source`
+        #[allow(clippy::unnecessary_safety_comment)]
```

Comment:
This is an annoying rule it seems, I'll remove it tomorrow.

Reply:
I think it's good to leave it enabled. It's helpful when you convert an unsafe function to safe, as clippy tells you everywhere you can now remove the unsafe blocks and comments.

It can be disabled on a module-by-module basis where it comes up frequently.